engine/src/main/java/graphql/nadel/engine/execution/AsyncIsFieldForbidden.java

```diff
@@ -22,13 +22,20 @@
 
     private final Map<NormalizedQueryField, GraphQLError> fieldsToErrors = new ConcurrentHashMap<>();
     private final ServiceExecutionHooks serviceExecutionHooks;
-    private final ExecutionContext executionContext;
     private final NadelContext nadelContext;
+    private final HydrationArguments hydrationArguments;
+    private final Map<String, Object> variables;
 
-    public AsyncIsFieldForbidden(ServiceExecutionHooks serviceExecutionHooks, ExecutionContext executionContext, NadelContext nadelContext) {
+    public AsyncIsFieldForbidden(
```

Comment:
lets use the new `nonNullable/nullable` annotations here to indicate what could be null or not...

Reply:
added `org.jetbrains.annotations.NotNull` to contructor args, fields, and the public method